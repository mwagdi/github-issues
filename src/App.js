import React from 'react';
import {
    BrowserRouter,
    Switch,Route } from 'react-router-dom';

import Topbar from './components/Topbar';
import Issues from './routes/Issues';
import SingleIssue from './routes/SingleIssue';
import IssuesHeader from './components/IssuesHeader';
import Footer from './components/Footer';
import './assets/scss/app.scss';

const App = props => (
    <BrowserRouter>
        <div className="app flex-container flex-column">
            <Topbar />
            <div className="issues flex-grow">
                <IssuesHeader />
                <Switch>
                    <Route path="/:number" component={SingleIssue} />
                    <Route path="/" component={Issues} />
                </Switch>
            </div>
            <Footer />
        </div>
    </BrowserRouter>
);

export default App;