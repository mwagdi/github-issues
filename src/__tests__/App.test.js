import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import App from '../App';


test('App renders correctly', () => {
    const wrapper = shallow(<App />);
    expect(toJSON(wrapper)).toMatchSnapshot();
    
})