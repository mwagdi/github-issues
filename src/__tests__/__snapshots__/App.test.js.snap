// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<BrowserRouter>
  <div
    className="app flex-container flex-column"
  >
    <Topbar />
    <div
      className="issues flex-grow"
    >
      <IssuesHeader />
      <Switch>
        <Route
          component={[Function]}
          path="/"
        />
      </Switch>
    </div>
    <_default />
  </div>
</BrowserRouter>
`;
